test/test_pebble.py

```diff
@@ -1076,6 +1161,423 @@ def test_get_services_names(self):
+        self.assertEqual(infos[0].permissions, 0o644)
+        self.assertEqual(infos[0].last_modified, datetime_nzdt(2021, 1, 28, 14, 37, 4, 291518))
+        self.assertEqual(infos[0].user_id, 12)
+        self.assertEqual(infos[0].group_id, 34)
+        self.assertEqual(infos[1].path, '/etc/nginx')
+        self.assertEqual(infos[1].name, 'nginx')
+        self.assertEqual(infos[1].type, pebble.FileType.DIRECTORY)
+        self.assertEqual(infos[1].size, None)
+        self.assertEqual(infos[1].permissions, 0o755)
+        self.assertEqual(infos[1].last_modified, datetime_nzdt(2020, 1, 1, 1, 1, 1, 0))
+        self.assertIs(infos[1].user_id, None)
+        self.assertIs(infos[1].group_id, None)
+
+        self.assertEqual(self.client.requests, [
+            ('GET', '/v1/files', {'action': 'list', 'path': '/etc'}, None),
+        ])
+
+    def test_list_files_pattern(self):
+        self.client.responses.append({
+            "result": [],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+
+        infos = self.client.list_files('/etc', pattern='*.conf')
+
+        self.assertEqual(len(infos), 0)
+        self.assertEqual(self.client.requests, [
+            ('GET', '/v1/files', {'action': 'list', 'path': '/etc', 'pattern': '*.conf'}, None),
+        ])
+
+    def test_list_files_itself(self):
+        self.client.responses.append({
+            "result": [],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+
+        infos = self.client.list_files('/etc', itself=True)
+
+        self.assertEqual(len(infos), 0)
+        self.assertEqual(self.client.requests, [
+            ('GET', '/v1/files', {'action': 'list', 'path': '/etc', 'itself': 'true'}, None),
+        ])
+
+    def test_make_dir_basic(self):
+        self.client.responses.append({
+            "result": [{'path': '/foo/bar'}],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+        self.client.make_dir('/foo/bar')
+        dir_item = {
+            'path': '/foo/bar',
+        }
+        self.assertEqual(self.client.requests, [
+            ('POST', '/v1/files', None, {'action': 'make-dirs', 'dirs': [dir_item]}),
+        ])
+
+    def test_make_dir_full(self):
+        self.client.responses.append({
+            "result": [{'path': '/foo/bar'}],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+        self.client.make_dir('/foo/bar', make_parents=True, permissions=0o600,
+                             user=12, group=34)
+
+        dir_item = {
+            'path': '/foo/bar',
+            'make-parents': True,
+            'permissions': '600',
+            'user-id': 12,
+            'group-id': 34,
+        }
+        self.assertEqual(self.client.requests, [
+            ('POST', '/v1/files', None, {'action': 'make-dirs', 'dirs': [dir_item]}),
+        ])
```

Comment:
I didn't review all of these tests with a fine tooth comb. If there are any that you have concerns about, I'm happy to focus more on them.

Reply:
I don't have any concerns about them. They're very straight-forward as request and responses are mocked.